docs/configuration.md

```diff
@@ -1,5 +1,257 @@
 ## Introduction
-All of the configuration files for Lychee are stored in the config directory. Each option is documented, so feel free to look through the files and get familiar with the options available to you.
+
+Configuration is managed using a `.env` file. It probably exists already in your directory (Composer copies it from `.env.example` during installation), and you can go on with configuring there. But the file does not contain all options. Here, you can find all of them in a structured way and with instructions on how to use them. All of the options are optional, unless you would like to use some services.
```

Comment:
I'm not in the headspace right now, but this needs tweaking.

Reply:
Someone else should proofread this.
```suggestion
Lychee's core configuration is managed using a `.env` file. It probably exists already in your directory, but if not it can be created by copying `.env.example`. Most of the options are already included there, but this page contains a more complete list with descriptions and default values.
```
Doesn't Lychee fail to run without `APP_KEY`, making that one at least mandatory?

Reply:
> If Composer runs correctly, it is set already.

If deleting it breaks things, that isn't optional.

Reply:
I agree with @d7415, but I don't understand how the discussion is related to the referenced text. Neither the previous nor the suggested text says anything about configuration options which are optional. What am I missing here?

Reply:
@nagmat84 In the original patch:
> All of the options are optional, unless you would like to use some services.

Reply:
Silly me, it is in the previous text:

> **All of the options are optional,** unless you would like to use some services

Yes this is indeed a wrong statement. @d7415 is right. This does not only hold for `APP_KEY` but also the database configuration. Although Lychee comes with a sensible default and configures an SQLite file-based database, one database must always be configured. So this is not optional either.

@qwerty287 Is it possible that you wanted rather say something like this: "All advanced options which are listed here but are not explicitly included in the default `.env` file are optional unless you would like or are obliged to use the services they belong to."

Again this sentence needs tweaking, but I am not sure whether we want to says something like this at all. In the end, this is a tautology. Something is always optional unless it is required. For example, let's take the `TRUSTED_PROXY` variable. For 85% of the users this is probably an optional setting. 10% may use it voluntarily, because the like to have fancy setup for the sake of it. But there may also be 5% who must use this option ("are obliged to use it"), because their hoster enforces such a setup.

Reply:
Here is another suggestion, during dinner I got an idea what @qwerty287 might have meant:

_Lychee's core configuration is managed using a `.env` file. It probably exists already in your directory, but if not it can be created by copying `.env.example`. The options which are already included in the `.env` file should be sufficient to cover the necessary configuration for the vast majority of all use-cases and setups. Using other options than those included in the `.env` file should rarely be necessary. However, this page contains a  more complete list of the available options, incl. some highly advanced ones, together with descriptions and default values._

Reply:
That works, but feels very verbose.

Reply:
@qwerty287 Please fix this, this seems to be the last open issue in the his PR. Maybe simply use my last suggestion. @d7415 already said that it works even though it is verbose. But at least it is correct in content.